File: enaFrontend/src/app/models/eleve.ts
import { Adresse } from "./adresse";
import { Dossier } from "./dossier";
import { EleveCycle } from "./eleve-cycle";
import { ElevePromotion } from "./eleve-promotion";
import { EleveSection } from "./eleve-section";
import { Telephone } from "./telephone";
import { Urgence } from "./urgence";

export class Eleve {
    public id!: number;
    public nom!: string;
    public prenom!: string;
    public dateNaiss!: Date;
    public sexe!: string;
    public nci!: string;
    public typeEleve!: string;
    public lieuNaiss!: string;
    public nationalite!: string;
    public matricule!: string;
    public numDosCandidat!: string;
    public emailPrive!: string;
    public emailPro!: string;
    public prenomPere!: string;
    public prenomMere!: string;
    public nomPere!: string;
    public nomMere!: string;
    public ville!: any;
    public adresses!: Adresse[];
    public telephones!: Telephone[];
    public dossiers!: Dossier[];
    public urgences!: Urgence[];
    public eleveSections!: any;
    public eleveCycles!: any;
    public elevePromotions!: any;
}

File: ena/src/main/java/com/ena/entities/Eleve.java
package com.ena.entities;

import com.ena.deserializer.EleveDeserializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString
@JsonDeserialize(using = EleveDeserializer.class)
public class Eleve implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nom;
    private String prenom;
    private Date dateNaiss;
    private String sexe;
    private String nci;
    private String typeEleve;
    private String lieuNaiss;
    private String nationalite;
    private String matricule;
    private String numDosCandidat;
    private String emailPrive;
    private String emailPro;
    private String prenomPere;
    private String nomPere;
    private String prenomMere;
    private String nomMere;
    @ManyToOne
    private Ville ville;
    @OneToMany(mappedBy = "eleve")
    private List<Adresse> adresses;
    @OneToMany(mappedBy = "eleve")
    private List<Telephone> telephones;
    @OneToMany(mappedBy = "eleve")
    private List<Urgence> urgences;
    @OneToMany(mappedBy = "eleve")
    private List<EleveCycle> eleveCycles;
    @OneToMany(mappedBy = "eleve")
    private List<ElevePromotion> elevePromotions;
    @OneToMany(mappedBy = "eleve")
    private List<EleveSection> eleveSections;
    @OneToMany(mappedBy = "eleve")
    private List<Dossier> dossiers;

    public void addEleveCycle(EleveCycle eleveCycle) {
        if (!eleveCycles.contains(eleveCycle)) {
            eleveCycles.add(eleveCycle);
        }
    }

    public void removeEleveCycle(EleveCycle eleveCycle) {
        if (eleveCycles.contains(eleveCycle)) {
            eleveCycles.remove(eleveCycle);
        }
    }

    public void addEleveSection(EleveSection eleveSection) {
        if (!eleveSections.contains(eleveSection)) {
            eleveSections.add(eleveSection);
        }
    }

    public void removeEleveSection(EleveSection eleveSection) {
        if (eleveSections.contains(eleveSection)) {
            eleveSections.remove(eleveSection);
        }
    }

    public void addElevePromotion(ElevePromotion elevePromotion) {
        if (!elevePromotions.contains(elevePromotion)) {
            elevePromotions.add(elevePromotion);
        }
    }

    public void removeElevePromotion(ElevePromotion elevePromotion) {
        if (elevePromotions.contains(elevePromotion)) {
            elevePromotions.remove(elevePromotion);
        }
    }
}
